internal/models/validator.go

```diff
@@ -0,0 +1,23 @@
+package models
+
+import (
+	"context"
+
+	"github.com/go-playground/validator/v10"
+	"golang.org/x/crypto/ssh"
+)
+
+var validate *validator.Validate
+
+func init() {
+	validate = validator.New()
+
+	_ = validate.RegisterValidation("sshPubkey", func(fl validator.FieldLevel) bool {
+		_, err := ssh.ParsePublicKey(fl.Field().Bytes())
+		return err == nil
+	})
+}
+
+func Validate(ctx context.Context, model interface{}) error {
+	return validate.StructCtx(ctx, model)
```

Comment:
I see that we are validating a model using this method. 
According to the documentation, we are validating the exposed fields and by passing the ctx contextual validation information happens. but against what do the validations occur?

Reply:
I've added the `validate` tags, on the model structure fields, so there we define what rule should be used to validate given field, the validator library automatically extracts that tag and uses those rules. I dig its golang native feel, but there are libraries that allow more explicit validations and I'm happy to talk about those if we do not like this :)

Reply:
My question is, what rules do the tags validate?

Reply:
This PR is adding `sshPubkey` for which you can read the definition in the method, rest is defined in the library here: https://github.com/go-playground/validator#baked-in-validations
I'm using just required in this PR, but we can obviously use any validation in the future :)

Reply:
yup, you are just making the list of validations in this manner: https://github.com/RHEnVision/provisioning-backend/pull/106/files#diff-645fe10c68e235c13074b5db21d4033bbf06828ebd1fa00ad5fb1c336f52e24fR15